class/subscription.php

```diff
@@ -959,23 +989,45 @@ function force_pack_permission( $perm, $id, $form_settings ) {
         $form         = new WPUF_Form( $id );
         $force_pack   = $form->is_enabled_force_pack();
         $pay_per_post = $form->is_enabled_pay_per_post();
+        $fallback_enabled  = $form->is_enabled_fallback_cost();
+        $fallback_cost     = $form->get_subs_fallback_cost();
+
         $current_user = wpuf_get_user();
+        $current_pack = $current_user->subscription()->current_pack();
+        $has_post_count = $current_user->subscription()->has_post_count( $form_settings['post_type'] );
 
         if ( is_user_logged_in() ) {
 
             if ( wpuf_get_user()->post_locked() ) {
                 return 'no';
             } else {
+
+                // if post locking not enabled
                 if ( !$form->is_charging_enabled() ) {
                     return 'yes';
                 } else {
-                    if ( $force_pack ) {
-                        if ( !$current_user->subscription()->current_pack_id() ) {
+                    //if charging is enabled
+                    if ( $force_pack ) { 
+                        if ( ! is_wp_error( $current_pack ) ) {
+                            if ( ! $fallback_enabled ) {
+                                if ( !$current_user->subscription()->current_pack_id() ) {
```

Comment:
Maybe this complex if/else checking needs to move in a separate function, or to the users subscription class.